app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class UsersController < API::V1::ApplicationController
+      include API::V1::ErrorHandlerConcern
+
+      skip_before_action :doorkeeper_authorize!, only: :create
+
+      def create
+        user = User.new(create_params.except(:client_id))
+        client_app = Doorkeeper::Application.find_by(uid: create_params[:client_id])
+
+        unless client_app
+          render_error detail: 'Invalid client ID', status: :forbidden, source: :client_id
+          return
+        end
+
+        if user.save
+          # create access token for the user, so the user won't need to login again after registration
+          access_token = user.get_access_token client_app.id
+          # return json containing access token and refresh token
+          # so that user won't need to call login API right after registration
+          render json: create_success_result(user, access_token), status: :created
+
+        else
+          render_errors user.jsonapi_errors
```

Comment:
I added a method to the model "Users" (via error_handler_concern) in order to parse the model 'errors' into a Json::API compliant error. 
But I'm not sure if this is the right place to be done... (JSON::API is a controller responsibility, not much on the model // but parsing ActiveModel::Error objects seems to be the responsibility of the model... humhum

Any insight @junan  @ankitkalia1195  ? :)
Thanks!

Reply:
@malparty  We usually don't place API related stuff into model. 

I think you can get the related errors with ` user.errors.full_messages.to_sentence`. You can render the error like we did [Survey project](https://github.com/nimblehq/nimble-survey-web/blob/development/app/controllers/api/v1/registrations_controller.rb#L13)
```ruby
  render_error(detail: user.errors.full_messages.to_sentence)
```

Reply:
full messages merge all errors into a string, which is a bit sad if we need to assign it to "errors'. 
I'll figure out to have it cleaned :)

Reply:
@junan  Okay, code updated, I just created a simple ActiveModel::ErrorsSerializer :) 
So that I can `ErrorsSerializer.new(user.errors).serializable_hash`. 

It seems clean to me~